src/main/kotlin/exercise2/task2/FindPairOfHighestSumFunctional.kt

```diff
@@ -20,7 +20,9 @@ import org.jetbrains.exercise2.task3.findPairWithBiggestDifference
  */
 
 internal fun List<Int>.findHighestSumPairFunctional(): Pair<Int, Int> {
-    TODO("Implement me!!")
+    return Pair(
+        this.sorted().get(this.lastIndex), this.sorted().get(this.lastIndex-1)
```

Comment:
This is not a functional approach :) 

The idea of the functional approach is that you compose function calls that operate on data to get the desired result. 

Functional approach would be following:
```suggestion
        return this.sorted().let { sortedList -> Pair(this.lastIndex, this.lastIndex-1)  }
```
In this case, we are using the result of the `sorted` function invocation, and we are providing it as an input of the `let` function invocation. Hence, we are composing function calls to get the result.